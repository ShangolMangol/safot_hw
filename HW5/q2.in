use_module(library(yall)).
print("===TEST START===").

legal_graph(graph([[2, 3], [], [2, 4], [3]])).
legal_graph(graph([])).

{L}/(
    findall(
        X,
        edge(graph([[2, 3], [], [2, 4], [3]]), X, 3),
        L1
    ),
    msort(L1, L)
).
edge(graph([[2, 3], [], [2, 4], [3]]), 2, X).

{L}/(
    findall(
        P,
        path(graph([[2, 3], [], [2, 4], [3]]), 1, 4, P),
        L1
    ),
    msort(L1, L)
).
path(graph([[2, 3], [], [2, 4], [3]]), 2, 4, _).

{L}/(
    findall(
        P,
        circle(graph([[2, 3], [], [2, 4], [3]]), P),
        L1
    ),
    msort(L1, L)
).

dag(graph([[2, 3], [], [2, 4], [3]])).
dag(graph([[2, 3], [], [2], [3]])).

