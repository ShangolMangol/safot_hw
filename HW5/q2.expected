
true.

false.

L = [1, 4].

false.

L = [[1, 3, 4]].

false.

L = [[3, 4, 3], [4, 3, 4]].

false.

true.


